crates/matrix-sdk-crypto/src/machine/mod.rs

```diff
@@ -1570,11 +1567,69 @@ impl OlmMachine {
         self.get_encryption_info(&session, &event.sender).await
     }
 
+    /// Check whether the sender of a Megolm session is trusted, based on the
+    /// verification state.
+    ///
+    /// This is used by `check_sender_trust_requirement`, and ensures that the
+    /// sending device is cross-signed.
+    fn check_sender_trusted(&self, encryption_info: &EncryptionInfo) -> MegolmResult<()> {
```

Comment:
per a discussion today in `#matrix-rust-sdk-dev`, helper functions should probably come after the functions they help.